app/assets/javascripts/admin/addon/controllers/admin-embedding-posts-and-topics.js

```diff
@@ -0,0 +1,46 @@
+import Controller, { inject as controller } from "@ember/controller";
+import { action } from "@ember/object";
+import { service } from "@ember/service";
+import { isEmpty } from "@ember/utils";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+import { i18n } from "discourse-i18n";
+
+export default class AdminEmbeddingPostsAndTopicsController extends Controller {
+  @service toasts;
+  @controller adminEmbedding;
+
+  get formData() {
+    const embedding = this.adminEmbedding.embedding;
+
+    return {
+      embed_by_username: isEmpty(embedding.embed_by_username)
+        ? null
+        : [embedding.embed_by_username],
+      embed_post_limit: embedding.embed_post_limit,
+      embed_title_scrubber: embedding.embed_title_scrubber,
+      embed_truncate: embedding.embed_truncate,
+      embed_unlisted: embedding.embed_unlisted,
+    };
+  }
+
+  @action
+  async save(data) {
+    const embedding = this.adminEmbedding.embedding;
+
+    try {
+      await embedding.update({
+        type: "posts_and_topics",
+        ...data,
+        embed_by_username: data.embed_by_username[0],
```

Comment:
Same question as another comment, is there something special about the user at position 0 or should current user be used or something?

Reply:
Same as previous
<img width="497" alt="Screenshot 2024-12-20 at 8 45 31 AM" src="https://github.com/user-attachments/assets/12600ef4-82ff-42d2-93eb-eb9608c791c4" />

```
    <field.Custom>
      <UserChooser
        @value={{field.value}}
        @onChange={{field.set}}
        @options={{hash maximum=1 excludeCurrentUser=false}}
        class="admin-embedding-posts-and-topics-form__embed_by_username"
      />
    </field.Custom>
```
I considered changing the backend to accept both, arrays and single values, but decided that the frontend should send the correct data. Backend accepting array even if only one value is allowed would be even weirder. 

Before our redesign, these fields were just text input so the user selector is an improvement.

Before:
<img width="560" alt="Screenshot 2024-12-20 at 8 48 59 AM" src="https://github.com/user-attachments/assets/9ca822a3-b90f-47a0-953e-e36e6f7d74ba" />

Reply:
Ah ok fair, sorry didn't realise